grant-project/zcash-grant-system
# Allow topping off of partial stake contributions. Fix float innaccura…

Closes #141, closes #157.

### What This Does

* Adds a "Stake" button to pending staking proposals on the profile that either fetches the existing stake contribution, or creates a new "top-off" stake with the remaining amount.
* Switch from `float` to `Decimal` wherever possible to avoid various bugs.
* Modifies the partial contribution email to indicate how to top off the stake.
* Allows staking proposals to be deleted.

### Steps to Test

* Create a proposal and submit it. Skip the staking step.
* Go to your profile's pending tab. Click "Stake" on the staking proposal. Confirm it shows the same contribution.
* Send a partial amount to it. Confirm you get an email about the partial contribution.
* Go back to your profile's pending tab, and click "Stake" again. Confirm you're given a new contribution with the amount remaining.
* Send the new contribution and confirm you're no longer in staking state afterwards.

## Screenshots

<img width="500" alt="screen shot 2019-02-05 at 8 53 31 pm" src="https://user-images.githubusercontent.com/649992/52316137-1ec48f00-2988-11e9-84ee-287e56bf10c0.png">


<img width="300" alt="screen shot 2019-02-05 at 8 53 23 pm" src="https://user-images.githubusercontent.com/649992/52316131-1c623500-2988-11e9-88b7-b3786e687afe.png">
